### PR TITLE
📖 document migrating to user-managed cert-manager

### LINKED ISSUE #7280: document migrating to user-managed cert-manager
**User Story**

As an operator I would like to ignore cert-manager during upgrades so I can self-manage cert-manager deployments.

**Detailed Description**
Currently when upgrading CAPI through clusterctl, it tries to upgrade cert-manager based on the annotation `cert-manager.clusterctl.cluster.x-k8s.io/version`, as we manage our own cert-manager deployment this annotation is almost always out of date.

**Anything else you would like to add:**
Ideally, there would be an option to ignore cert-manager entirely through the upgrade process, as we currently handle the cert-manager upgrade process independently of CAPI. Maybe something like this?
```yaml
cert-manager:
  ignore: true
```
Another alternative could be looking at the label `app.kubernetes.io/version` instead of the annotation `cert-manager.clusterctl.cluster.x-k8s.io/version` and using that to determine an upgrade, that way it isn't tied to clusterctl?

/kind feature

### PR DESCRIPTION
**Pull request by charlie-haley**
**What this PR does / why we need it**:
Documents migrating from a `clusterctl` managed cert-manager to user-managed.

**Which issue(s) this PR fixes**
Fixes #7280

### CODE REVIEW

File: docs/book/src/tasks/migrating-to-user-managed-cert-manager.md
Comment (killianmuldoon) [inline docs/book/src/tasks/migrating-to-user-managed-cert-manager.md:9]: Would moving or repeating this line, or something like it, after the above command be clearer?

`If you want to manage and install your own cert-manager, you'll need to remove this label from all API resources.`

File: docs/book/src/SUMMARY.md
Comment (killianmuldoon) [inline docs/book/src/SUMMARY.md:35]: This might be better organised if associated with other clusterctl cert management configuration topics here: https://cluster-api.sigs.k8s.io/clusterctl/configuration.html#cert-manager-configuration

**Review by killianmuldoon — APPROVED** [review]
Thanks!

/lgtm

**Comment by sbueringer** [comment]
/lgtm

@fabriziopandini Please don't approve the GitHub actions. I would use this PR to test if the linter is now enforced

/hold
(just for my experiment. I'll drop it once the PR is approved from Fabrizio)

File: docs/book/src/clusterctl/configuration.md
Comment (fabriziopandini) [inline docs/book/src/clusterctl/configuration.md:102]:
```suggestion
`clusterctl` looks for the label `clusterctl.cluster.x-k8s.io/core=cert-manager` on all api resources in the `cert-manager` namespace. If it finds the label, `clusterctl` will manage the cert-manager deployment. You can list all the resources with that label by running:
```

**Comment by sbueringer** [comment]
Looks good, thx!

/lgtm
/approve

**Comment by sbueringer** [comment]
@charlie-haley PR will be merged soon. I'll just want to take a deep-dive into the Prow merge-bot to figure out why it doesn't behave as expected. I'll merge afterwards

**Review by fabriziopandini — COMMENTED** [review]
/approve

**Comment by sbueringer** [comment]
@fabriziopandini Please don't remove the hold. Debugging tide..

**Comment by sbueringer** [comment]
Okay after debugging through tide. 

The PR should not be merged. It's just that the missing linter is not shown in the Tide UI

/hold cancel

PR should now be stuck as the lint context is not successful even though it's required. Let's see what happens

**Comment by sbueringer** [comment]
(allowed the linters, so everything behaving as expected now)

**Pull request merged**
